drivers/serial/uart_stm32.c

```diff
@@ -455,7 +456,12 @@ static void uart_stm32_poll_out(const struct device *dev,
 	while (!LL_USART_IsActiveFlag_TXE(UartInstance)) {
 	}
 
+#ifdef CONFIG_PM
+	/* do not allow system to suspend until transmission has completed */
+	pm_constraint_set(PM_STATE_SUSPEND_TO_IDLE);
+#else
 	LL_USART_ClearFlag_TC(UartInstance);
+#endif /* CONFIG_PM */
```

Comment:
Minor enhancement proposal:
According to RM, use of TC is "to avoid corrupting the last transmission when the USART is
disabled or enters Halt mode.".
Since we're not checking it when PM=n, I think it should be safe removing it completely from here.

Reply:
done